src/compact.ts

```diff
@@ -0,0 +1,49 @@
+import { Kind } from '@stoplight/yaml-ast-parser';
+import { YAMLAnchorIgnoreKeys, YAMLNode, YAMLScalarIgnoreKeys } from './types';
+import { isObject } from './utils';
+
+const ANCHOR_IGNORE_KEYS: YAMLAnchorIgnoreKeys[] = ['referencesAnchor', 'anchorId'];
+const SCALAR_IGNORE_KEYS: YAMLScalarIgnoreKeys[] = ['doubleQuoted', 'singleQuoted', 'plainScalar', 'rawValue'];
+
+export function compact(node: YAMLNode): void {
```

Comment:
Urgh, I really wish we didn't need to traverse the tree from the very top to the very bottom yet again.
Unfortunately, nodes need to be retained for the entire process of parsing, as they might be referencing each other (anchors), and if they do so, we might be unable to produce proper values for scalar nodes as some of the required properties could be swept.
Obviously, we could perform a lookup to see what value was generated for the pristine node, but this would make everything more complicated.